firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -174,22 +175,30 @@ public static FirebaseAppDistribution getInstance(@NonNull FirebaseApp app) {
   @NonNull
   public UpdateTask updateApp() throws FirebaseAppDistributionException {
 
```

Comment:
Feels a little like this update app logic is split between FirebaseAppDistribution class and UpdateAppClient. Why did we change the client method from `getUpdateTask` to `performUpdate`?

Reply:
It changed to performUpdate (now updateApp) because the UpdateTask gets new'd up in the FirebaseAppDistribution class and gets passed in rather than created in the client so we can do the null/iscomplete check before the signin and null release check, and we thought that its more clear that it is the updateTask passed in that is the one that is being used for listeners/setting result. We should chat about this